src/include/distributed/resource_lock.h

```diff
@@ -142,6 +144,73 @@ typedef enum CitusOperations
 						 (uint32) (taskId), \
 						 ADV_LOCKTAG_CLASS_CITUS_BACKGROUND_TASK)
 
+/*
+ * IsNodeWideObjectClass returns true if the given object class is node-wide,
+ * i.e., that is not bound to a particular database but to whole server.
+ *
+ * Defined here as an inlined function so that SET_LOCKTAG_GLOBAL_DDL_SERIALIZATION
+ * macro can use it.
+ */
+static inline bool
+IsNodeWideObjectClass(ObjectClass objectClass)
+{
+	if ((int) objectClass < 0 || objectClass > LAST_OCLASS)
+	{
+		elog(ERROR, "invalid object class: %d", objectClass);
+	}
+
+	/*
+	 * We don't expect Postgres to change an object class to a node-wide one in the
+	 * future, but a newly added object class may be node-wide.
+	 *
+	 * So we put a static assert here to make sure that the developer who adds support
+	 * for a new Postgres version is aware of this.
+	 *
+	 * If new object classes are added and none of them are node-wide, then update
+	 * this assertion check based on latest supported major Postgres version.
+	 */
+	StaticAssertStmt(PG_MAJORVERSION_NUM <= 16,
+					 "better to check if any of newly added ObjectClass'es are node-wide");
+
+	switch (objectClass)
+	{
+		case OCLASS_ROLE:
+		case OCLASS_DATABASE:
+		case OCLASS_SUBSCRIPTION:
+		case OCLASS_TBLSPACE:
+#if PG_VERSION_NUM >= PG_VERSION_15
+		case OCLASS_PARAMETER_ACL:
```

Comment:
hmm, this made me look what this was. Turns out we need to propagate some more global DDL that we don't support yet... GRANT/REVOKE ON PARAMETER